src/Device/Driver/AR62xx.cpp

```diff
@@ -347,15 +322,19 @@ unsigned int AR62xxDevice::GenerateSetStationCommand(uint8_t *command, unsigned
 
   /** setting frequencies */
   command[command_length++] = 22;
-  command[command_length++] = GetNthByte(active_frequency_idx, 1);
-  command[command_length++] = GetNthByte(active_frequency_idx, 0);
-  command[command_length++] = GetNthByte(passive_frequency_idx, 1);
-  command[command_length++] = GetNthByte(passive_frequency_idx, 0);
-
-  /** Creating the binary value */
-  uint16_t command_crc = CRCBitwise(command, command_length);
-  command[command_length++] = GetNthByte(command_crc, 1);
-  command[command_length++] = GetNthByte(command_crc, 0);
+
+  uint16_t tmp[] = {0x00, 0x00};
+  WriteUnalignedLE16((uint16_t *)(void *)tmp, active_frequency_idx);
+  command[command_length++] = tmp[1];
+  command[command_length++] = tmp[0];
```

Comment:
Did you test this with real hardware? Does it work?

Reply:
tested with live-system several times

Reply:
This cannot possibly work! If you tested this and it worked, it means something is awfully wrong with the way you test this driver.
Look at what happens here: you have a variable `tmp` which is two `uint16_t` (i.e. totalling 4 bytes); the second one is default-initialized to zero and is never changed. The first one is written to by `WriteUnalignedLE16()`, writing 16 little-endian bits.
Now `command[command_length++] = tmp[1];` appends a zero byte to `command`.
And `command[command_length++] = tmp[0];` appends the lower byte of the frequency index to `command`.
This discards the upper byte of the frequency index!
Do you understand why everything about this piece of code is wrong? And why this cannot have worked properly? This piece of code makes no sense at all!

Reply:
Sorry, my explanation is outdated. In a previous version of this PR, `tmp` was an array of two `uint16_t`. After you corrected, this (maybe after my first comment here?), this can work. But is still bad, because it requires a temporary stack buffer for no reason.